### PR TITLE
Have integration tests running towards cluster set job names

### LINKED ISSUE #7727: Set job name to test name for cluster integration tests
In all integration tests involving the PBS or LSF cluster, we should set the name of the job equal to the test function.

This will aid in debugging because the job name is propagated through logs, finding the string "dummy" in the logs is not sufficient in order to figure out which test has failed f.ex. 

https://github.com/equinor/ert/blob/544caea887f9ca985787e5e81d3d90a6c63eddd2/tests/integration_tests/scheduler/test_generic_driver.py#L52

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #7727 

**Approach**
The commit in this PR makes the integration tests for the scheduler drivers use the test names as job names when doing `driver.submit`, making it easier to debug.

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
Does not work:

`ERROR    ert.scheduler.driver:driver.py:113 Command "/global/bin/bsub -o '/private/havb/projects/ert/delmee/test_submit_LsfDriver_0/test_submit[LsfDriver].LSF-stdout' -J 'test_submit[LsfDriver]' /private/havb/projects/ert/delmee/test_submit_LsfDriver_0/.lsf_submit_nrkkgciu.sh /private/havb/projects/ert/delmee/test_submit_LsfDriver_0" failed after 10 retries with error Bad job name. Job not submitted`

File: tests/integration_tests/scheduler/test_generic_driver.py [hunk 1]
```diff
@@ -73,13 +74,21 @@ async def finished(iens, returncode):
         nonlocal finished_called
         finished_called = True
 
-    await driver.submit(0, "sh", "-c", f"exit {expected_returncode}", runpath=tmp_path)
+    job_name: str = request.node.name.replace("[", "__").replace("]", "__")
```
Comment (berland) [inline tests/integration_tests/scheduler/test_generic_driver.py:77]: It might be possible to put this into a new fixture that can yield a jobname.

**Comment by jonathan-eq** [comment]
I have successfully tested the integration tests against real lsf cluster.

**Comment by berland** [comment]
> I have successfully tested the integration tests against real lsf cluster.

PBS?

**Comment by jonathan-eq** [comment]
> > I have successfully tested the integration tests against real lsf cluster.
> 
> PBS?

Yes, it passed with flying colors! 🥇

File: tests/integration_tests/scheduler/bin/bjobs.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def get_parser() -> argparse.ArgumentParser:
         description="Mocked LSF bjobs command reading state from filesystem"
     )
     parser.add_argument("jobs", type=str, nargs="*")
+    parser.add_argument("-w", action="store_true")
```
Comment (berland) [inline tests/integration_tests/scheduler/bin/bjobs.py:34]: Belongs to a different commit?

File: tests/integration_tests/scheduler/conftest.py [hunk 3]
```diff
@@ -4,10 +4,17 @@
...
     bin_path = Path(__file__).parent / "bin"
 
     monkeypatch.setenv("PATH", f"{bin_path}:{os.environ['PATH']}")
     monkeypatch.setenv("PYTEST_TMP_PATH", str(tmp_path))
     monkeypatch.setenv("PYTHON", sys.executable)
+
+
+@pytest.fixture
+def job_name(request) -> str:
+    return request.node.name.split("[")[0]
```
Comment (berland) [inline tests/integration_tests/scheduler/conftest.py:20]: Nice!

File: tests/integration_tests/scheduler/bin/bjobs.py (hunk 2 above)
Comment (jonathan-eq) [reply tests/integration_tests/scheduler/bin/bjobs.py:34]: No, it failed because of test_lsf_driver.py::test_job_name. When running towards the real cluster; it would shorten the job name, so I added `-w` to bjobs in the test. This then caused the mocked command to fail due to unknown flags.

**Review by berland — APPROVED** [review]
